src/main/java/fr/techad/edc/httpd/utils/TokenUtils.java

```diff
@@ -0,0 +1,75 @@
+package fr.techad.edc.httpd.utils;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTCreationException;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.auth0.jwt.interfaces.JWTVerifier;
+
+public class TokenUtils {
+	static final Logger LOGGER = LoggerFactory.getLogger(TokenUtils.class);
+	final String SECRET = "edc-server";
+	final String AUTH = "edc";
+
+	private static TokenUtils instance;
+	private String privateKey;
+	private FileUtils fileUtils;
+	private final String tokenPath = "./token.info";
+	private final String keyPath = "./private.key";
+
+	private TokenUtils() {
+		this.fileUtils = FileUtils.getInstance();
+	}
+
+	public static synchronized TokenUtils getInstance() {
+		if (instance == null)
+			instance = new TokenUtils();
+		return instance;
+	}
+
+	private String genSecretKey() {
+		return RandomStringUtils.randomAlphanumeric(24);
+	}
+
+	public boolean validateToken(String token) {
+		try {
+			Algorithm algorithm = Algorithm.HMAC256(SECRET);
+			JWTVerifier verifier = JWT.require(algorithm).withClaim("private", this.privateKey).withIssuer(AUTH)
```

Comment:
The privateKey can be null if the createTokenFile is not called.
Need to check and call the readPrivateKey.
If the key is empty, the token is unvalidated.